Userland/Utilities/xxd.cpp

```diff
@@ -0,0 +1,373 @@
+    }
+
+    while (is_input_remaining) {
+        auto bytes_to_read = max_read_size - bytes.size();
+
+        bytes = contents.span().slice(0, bytes_to_read);
+        bytes = TRY(file->read_some(bytes));
+
+        if (bytes.size() < bytes_to_read && file->fd() != STDIN_FILENO) {
+            is_input_remaining = false;
+        }
+
+        while (bytes.size() > 0) {
+            auto line_length = bytes.size() > line_length_config ? line_length_config : bytes.size();
+
+            if (max_bytes.has_value()) {
+                auto bytes_remaining = max_bytes.value() - total_bytes_read;
+                if (bytes_remaining < line_length) {
+                    line_length = bytes_remaining;
+                }
+            }
+
+            auto current_line = bytes.slice(0, line_length);
+            bytes = bytes.slice(line_length);
+
+            if (autoskip && all_of(bytes, [](auto& b) { return b == 0x0; })) {
+                outln("*");
+                continue;
+            }
+
+            if (display_style != DisplayStyle::CStyle && display_style != DisplayStyle::PlainHex) {
+                auto offset = 0;
+                if (position_offset.has_value()) {
+                    offset = position_offset.value();
+                }
+
+                if (offset_in_decimal) {
+                    out("{:08}: ", total_bytes_read + offset);
+                } else {
+                    out("{:08x}: ", total_bytes_read + offset);
+                }
+            }
+
+            switch (display_style) {
+            case DisplayStyle::Hex:
+                print_line_hex(current_line, line_length_config, group_size, uppercase_hex);
+                print_ascii(current_line);
+                break;
+            case DisplayStyle::PlainHex:
+                print_line_hex(current_line, line_length_config, group_size, uppercase_hex);
+                break;
+            case DisplayStyle::HexLittleEndian:
+                print_line_little_endian_hex(current_line, line_length_config, group_size, uppercase_hex);
+                print_ascii(current_line);
+                break;
+            case DisplayStyle::Bits:
+                print_line_bits(current_line, line_length_config, group_size);
+                print_ascii(current_line);
+                break;
+            case DisplayStyle::CStyle:
+                print_line_c_style(current_line);
+                break;
+            }
+
+            putchar('\n');
+
+            total_bytes_read += line_length_config;
+
+            if (max_bytes.has_value() && total_bytes_read >= max_bytes.value()) {
+                is_input_remaining = false;
+                break;
+            }
+        }
+    }
+
+    if (display_style == DisplayStyle::CStyle) {
+        outln("}};");
+        auto postfix = capitalize_c_include_file_style ? "LEN" : "len";
+        outln("size_t {}_{} = {};", c_include_file_style_variable_name, postfix, total_bytes_read);
```

Comment:
Same comment on C compliance as above. Also `total_bytes_read` doesn't seem to output the correct size when testing with my test file.
```suggestion
        outln("unsigned int {}_{} = {};", c_include_file_style_variable_name, postfix, total_bytes_read);
```